### ISSUE TITLE
Make it possible to use a linter only if another is unavailable

### ISSUE THREAD

**Issue by tkadur** [w0rp/ale#1126]
When writing C++, I want to use clang-check for linting whenever I'm in a project that has a compile_commands.json file and clang otherwise. If both are enabled, clang shows false errors in projects with the json file and clang-check shows false errors in projects without one. The only other issue I saw regarding this was #913 but that didn't end up being about the same thing. I scoured other issues and the help pages already but found no good built-in way to do this.

TL;DR: I want to use clang-check and clang mutually exclusively.

**Comment by w0rp** [comment]
You can configure linters for different files with buffer variables. See the `:help g:ale_pattern_options` for doing this easily based on regular expressions.

**Comment by tkadur** [comment]
I don't think this does quite what I want. Consider the following directory structure:

![image](https://user-images.githubusercontent.com/8755324/32976165-3856ab7c-cbdf-11e7-8e6c-fd8ba7cdcb0f.png)

The goal would be for lonely.cpp to use clang with the default options set for ALE but for main.cpp and fun.cpp to use clang-tidy with the compilation database provided there. Each option works perfectly well independently, but I have to manually change the ALE settings when editing each one.

**Comment by w0rp** [comment]
Use different regex that works.
